src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -1,751 +1,5624 @@
+            where T : IEquatable<T>?
+        {
+            return tensor.FlattenedLength == other.FlattenedLength
+                && tensor._shape.LinearLength == other._shape.LinearLength
+                && tensor.Lengths.SequenceEqual(other.Lengths)
+                && MemoryMarshal.CreateReadOnlySpan(in tensor.GetPinnableReference(), (int)tensor._shape.LinearLength).SequenceEqual(MemoryMarshal.CreateReadOnlySpan(in other.GetPinnableReference(), (int)other._shape.LinearLength));
+        }
+        #endregion
+
+        #region SetSlice
+        /// <summary>
+        /// Sets a slice of the given <paramref name="tensor"/> with the provided <paramref name="values"/> for the given <paramref name="ranges"/>
+        /// </summary>
+        /// <param name="tensor">Input <see cref="Tensor{T}"/>.</param>
+        /// <param name="values">The values you want to set in the <paramref name="tensor"/>.</param>
+        /// <param name="ranges">The ranges you want to set.</param>
+        public static Tensor<T> SetSlice<T>(this Tensor<T> tensor, in ReadOnlyTensorSpan<T> values, params ReadOnlySpan<NRange> ranges)
+        {
+            tensor.AsTensorSpan().SetSlice(values, ranges);
+            return tensor;
+        }
+
+        /// <summary>
+        /// Sets a slice of the given <paramref name="tensor"/> with the provided <paramref name="values"/> for the given <paramref name="ranges"/>
+        /// </summary>
+        /// <param name="tensor">Input <see cref="Tensor{T}"/>.</param>
+        /// <param name="values">The values you want to set in the <paramref name="tensor"/>.</param>
+        /// <param name="ranges">The ranges you want to set.</param>
+        public static ref readonly TensorSpan<T> SetSlice<T>(this in TensorSpan<T> tensor, scoped in ReadOnlyTensorSpan<T> values, params scoped ReadOnlySpan<NRange> ranges)
+        {
+            if (ranges.IsEmpty)
+            {
+                values.CopyTo(tensor);
+            }
+            else
+            {
+                values.CopyTo(tensor.Slice(ranges));
+            }
+            return ref tensor;
+        }
+        #endregion
+
+        #region Split
+        /// <summary>
+        /// Split a <see cref="Tensor{T}"/> into <paramref name="splitCount"/> along the given <paramref name="dimension"/>. If the tensor cannot be split
+        /// evenly on the given <paramref name="dimension"/> an exception is thrown.
+        /// </summary>
+        /// <param name="tensor">Input <see cref="Tensor{T}"/>.</param>
+        /// <param name="splitCount">How many times to split the <paramref name="tensor"/></param>
+        /// <param name="dimension">The axis to split on.</param>
+        public static Tensor<T>[] Split<T>(scoped in ReadOnlyTensorSpan<T> tensor, int splitCount, nint dimension)
+        {
+            if (tensor.Lengths[(int)dimension] % splitCount != 0)
```

Comment:
nit: do we need to check the `dimension` before using it? I know we'll get exception anyway if something goes wrong. Just to have the right informative exception.

Reply:
Good idea. Added locally for followup pr.